src/mon/MonmapMonitor.cc

```diff
@@ -231,6 +264,43 @@ void MonmapMonitor::dump_info(Formatter *f)
 }
 
 bool MonmapMonitor::preprocess_command(MonOpRequestRef op)
+{
+  using namespace ceph::mon_cmds;
+
+  assert(nullptr != mon->monmap);
+  
+  try 
+  {   
+      mon_cmd_ctx state { op, *this, *mon };
+
+      if (!commands.in_registry(state.prefix)) {
+          return legacy_preprocess_command(op);
+      }
+      
+      // Dispatch commands:
+      auto result = commands.apply(ceph::operation_type::preprocess, state.prefix, state);
+
+      if (mon_cmd_result::failure == *result) {
+          return false;
+      }
+
+      if (-1 == state.r) {
+          return false;
+      }
+
+      mon->reply_command(op, state.r,
+                         ceph::util::get_line(state.ss), state.rdata, get_last_committed());
+            
+      return true;
+  }
+  catch(mon_cmds::failure& e)
+  { 
+    mon->reply_command(op, e.error_code(), e.what(), get_last_committed());
```

Comment:
This answers my earlier question about getting strings out of failing commands, I think.